### ISSUE TITLE
MMLU computation

### ISSUE THREAD

**Issue by steremma** [artidoro/qlora#147]
Hello, i have a couple of questions regarding the MMLU score computations in the callback.

```
for batch in tqdm(data_loader, total=len(data_loader)):
    (loss, logits, labels) = trainer.prediction_step(trainer.model, batch, prediction_loss_only=False,)
    # There are two tokens, the output, and eos token.
    for i, logit in enumerate(logits):
        label_non_zero_id = (batch['labels'][i] != -100).nonzero()[0][0]
        logit_abcd = logit[label_non_zero_id - 1][abcd_idx]
        preds.append(torch.argmax(logit_abcd).item())
    labels = labels[labels != IGNORE_INDEX].view(-1, 2)[:, 0]
    refs += [abcd_idx.index(label) for label in labels.tolist()]
    loss_mmlu += loss.item()
```

Looking at the inner loop, I intepret as follows:
1. We loop over batch examples, let's say 8 of them. 
2. We check which target indices are not -100 (used both to mask input and to pad labels). 
3. Ideally `label_non_zero_id` should only contain 2 indices: one of ABCD and 0 (for EOS). We assume that and take the first
4. We check the 4 legit vocabulary logits for the **PREVIOUS** token (-1) and argmax

My 2 questions in order of importance:
1. Why that -1? Could this be a bug even?
2. If the model outputs invalid outputs (not A/B/C/D + EOS) shouldn't we consider it an error? E.g. now if it responds "breakfast is my favourite meal of the day" and the correct answer was B we will reward it in terms of accuracy
